docs/v6-migration.md

```diff
@@ -0,0 +1,103 @@
+:::
+
+### Support for .env file
+
+In v6 we've introduced native support for reading `.env` files. You can use this file to protect your API keys and passwords. Usage of this feature is entirely optional but encouraged if you have the knowledge or experience with this type of configuration.
+
+The `.env` file needs to be present in the **current working directory** when you run cross-seed. In the case of the docker, the `.env` file needs to be inside the `/config` folder for the container to load it for you. For a `systemd` daemon, you set the `WorkingDirectory`.
+
+To access a value that is defined in the `.env` you use `process.env.VALUE`.
+
+Example:
+
+- `/config/.env`
+  ```
+  PROWLARR_API_KEY=abcd
+  QBIT_URL="http://username:password@qbittorrent:8080"
+  ```
+- `/config/config.js`
+  ```
+  module.exports = {
+    // ...
+    torznab: [
+      `http://prowlarr:9696/12/api?apikey=${process.env.PROWLARR_API_KEY}`
+    ],
+    // ...
+    qbittorrentUrl: process.env.QBIT_URL
+    // ...
+  }
+  ```
+
+:::note
+Any string with a variable inside of it should have the `process.env` reference wrapped with `${}` as well as enclosed in backticks (`) instead of quotes.
+:::
+
+### Other
+
+This section is not relevant for the usage, but merely for those interested in the other improvements we made. If you don't have any interest in development or don't care what happens behind the scenes to improve `cross-seed`, this probably is where you can stop reading.
+
+- We've updated to Node v20, ES2022, and TypeScript v5.
+
+- Indexer failures that are non-429 (rate limited) are now cleared from the database upon restarting.
+
+- We've fine-tuned the regexes we're using for matching pre-snatch. Specifically with the group matching. There were a few exceptions with trackers that have renamed their search results (releases) and non-standard naming for Arr libraries that we now attempt to account for.
+
+- Logging is now more granular for matching (matching decisions) - previously "file-tree mismatch" was used pretty ambiguously. This is now addressed.
+
+- There are now lists of files/folders integrated into `cross-seed` that are blocked during our prefiltering at startup. These include folders that exist in full-disc Bluray and DVD releases, as well as music files and rar archives, and season folders in Sonarr libraries, which were all previously searched individually. They will now be excluded from indexing your files (for data-based searching) and result in fewer "bad" searches that would otherwise yield no viable results.
+
+- We now have recommended defaults in the configuration template, these options are general guides for you to consider when configuring `cross-seed`. We in no way guarantee any of these settings or their consequences/effectiveness if you misuse `cross-seed` or break tracker rules. You should always read your tracker's rules before using `cross-seed``.

```

Comment:
```suggestion
### Other Miscellaneous Changes

Here is a short list of other changes made in v6. These are all behind-the-scenes updates made to improve `cross-seed`.

- Updated to Node v20, ES2022, and TypeScript v5
- Non-rate limited failures (status code `429`) will be cleared from the database when `cross-seed` is restarted
- Regex improvements. Some trackers rename search results or have non-standard naming conventions. The updated regex takes more of those into account and should find more matches.
- Improved logging messages, specifically around matching decisions.
- There are now lists of files/foldere=s integrated into `cross-seed` that are blocked during prefiltering at startup. These include folders present in full-disc Bluray/DVD releases, music files, rar archives, and season folders in Sonarr libraries. Excluding these from the `cross-seed` index (for data-based searches) will result in fewer "bad" searches that would otherwise yield no viable results.
- New recommended defaults in `config.template.js`. These settings are what we consider to be the best starting options when setting up `cross-seed`.
```